src/run_jack.rs

```diff
@@ -19,35 +22,33 @@ pub fn run_jack_generator(generator: Generator) -> Result<(), Error> {
         },
         generator,
     };
-    let _active_client = client.activate_async(notification_handler, process_handler)?;
-    sleep_forever();
-    Ok(())
-}
-
-fn sleep_forever() {
-    loop {
-        thread::sleep(time::Duration::new(100, 0));
-    }
+    client.activate_async(notification_handler, process_handler)
 }
 
 struct Stereo<Port> {
     left: Port,
     right: Port,
 }
 
-struct ProcessHandler_ {
+pub struct ProcessHandler_ {
     ports: Stereo<Port<AudioOut>>,
-    generator: Generator,
+    generator: Arc<Mutex<Generator>>,
 }
 
 impl ProcessHandler for ProcessHandler_ {
-    fn process(&mut self, _client: &Client, scope: &ProcessScope) -> Control {
-        let left_buffer: &mut [f32] = self.ports.left.as_mut_slice(scope);
-        let right_buffer: &mut [f32] = self.ports.right.as_mut_slice(scope);
-        self.generator
-            .generate(_client.sample_rate() as i32, left_buffer);
-        for sample_index in 0..right_buffer.len() {
-            right_buffer[sample_index] = left_buffer[sample_index];
+    fn process(&mut self, client: &Client, scope: &ProcessScope) -> Control {
+        match self.generator.lock() {
+            Ok(mut generator) => {
+                let left_buffer: &mut [f32] = self.ports.left.as_mut_slice(scope);
+                let right_buffer: &mut [f32] = self.ports.right.as_mut_slice(scope);
+                generator.generate(client.sample_rate() as i32, left_buffer);
+                for sample_index in 0..right_buffer.len() {
+                    right_buffer[sample_index] = left_buffer[sample_index];
+                }
```

Comment:
Is there a better way to copy `left_buffer` into `right_buffer`? @casey @nebkor

Reply:
I think you can do:

```rust
let left_buffer: &mut [f32] = self.ports.left.as_mut_slice(scope);
generator.generate(client.sample_rate() as i32, left_buffer);
self.ports.right.as_mut_slice(scope).copy_from_slice(left_buffer);
```

Which should compile down to a memcpy.

Reply:
Yes, that works, thanks!